[java] Mark obsolete variables deprecated in DriverCommand and W3CHttpCommandCodec

iampopovich opened the pull request:
### **User description**
<!-- Thanks for Contributing to Selenium! -->
<!-- Please read our contribution guidelines: https://github.com/SeleniumHQ/selenium/blob/trunk/CONTRIBUTING.md -->

### 🔗 Related Issues
Related to #10397

### 💥 What does this PR do?
changes to remove driver commands are the same as in the [commit](https://github.com/SeleniumHQ/selenium/commit/24a5741427640f92a48590e838b05d93da0a7196) 
also rb and node bindings need similar changes 

### 🔧 Implementation Notes
This pull request removes support for local storage and session storage commands from the Selenium WebDriver's W3C HTTP Command Codec. The changes streamline the codebase by eliminating unused or deprecated commands and their associated logic.

### Removal of Local Storage and Session Storage Commands:

* [`java/src/org/openqa/selenium/remote/DriverCommand.java`](diffhunk://#diff-0ebc8850a70ad45102db2f0b1d321b78fb19228324c424b8685ce244eb0239f9L109-L120): Removed constants for local storage and session storage commands, such as `GET_LOCAL_STORAGE_ITEM`, `SET_LOCAL_STORAGE_ITEM`, and `CLEAR_SESSION_STORAGE`.
* [`java/src/org/openqa/selenium/remote/codec/w3c/W3CHttpCommandCodec.java`](diffhunk://#diff-d3ee2e9bb5279f45d35691b5fe24402e8b9d1b795b293ace7a5e4c4ec144d5f4L23-L24): Removed import statements for local storage and session storage commands that were no longer needed. [[1]](diffhunk://#diff-d3ee2e9bb5279f45d35691b5fe24402e8b9d1b795b293ace7a5e4c4ec144d5f4L23-L24) [[2]](diffhunk://#diff-d3ee2e9bb5279f45d35691b5fe24402e8b9d1b795b293ace7a5e4c4ec144d5f4L50-L70)
* [`java/src/org/openqa/selenium/remote/codec/w3c/W3CHttpCommandCodec.java`](diffhunk://#diff-d3ee2e9bb5279f45d35691b5fe24402e8b9d1b795b293ace7a5e4c4ec144d5f4L120-L131): Removed alias definitions for local storage and session storage commands, which mapped these commands to JavaScript execution.
* [`java/src/org/openqa/selenium/remote/codec/w3c/W3CHttpCommandCodec.java`](diffhunk://#diff-d3ee2e9bb5279f45d35691b5fe24402e8b9d1b795b293ace7a5e4c4ec144d5f4L221-L268): Removed case statements and associated JavaScript logic for handling local storage and session storage commands. This includes commands like `CLEAR_LOCAL_STORAGE`, `GET_SESSION_STORAGE_KEYS`, and `SET_SESSION_STORAGE_ITEM`.

### 💡 Additional Considerations
<!--- Are there any decisions that need to be made before accepting this PR? -->
<!--- Is there any follow-on work that needs to be done? (e.g., docs, tests, etc.) -->

### 🔄 Types of changes
<!-- ✂️ Please delete anything that doesn't apply -->
- Cleanup (formatting, renaming)


___

### **PR Type**
Bug fix


___

### **Description**
- Remove deprecated HTML5 local storage and session storage commands

- Clean up unused driver command constants and imports

- Eliminate associated command codec aliases and implementations

- Streamline W3C HTTP command codec for better maintainability


___

### **Changes diagram**

```mermaid
flowchart LR
  A["DriverCommand.java"] -- "Remove constants" --> B["Storage Commands Removed"]
  C["W3CHttpCommandCodec.java"] -- "Remove imports" --> B
  C -- "Remove aliases" --> B
  C -- "Remove implementations" --> B
  B --> D["Cleaner Codebase"]
```


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Cleanup</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>DriverCommand.java</strong><dd><code>Remove HTML5 storage command constants</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/remote/DriverCommand.java

<li>Removed 12 local storage command constants<br> <li> Removed 6 session storage command constants<br> <li> Cleaned up interface definition


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16013/files#diff-0ebc8850a70ad45102db2f0b1d321b78fb19228324c424b8685ce244eb0239f9">+0/-12</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>W3CHttpCommandCodec.java</strong><dd><code>Remove storage command codec implementations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

java/src/org/openqa/selenium/remote/codec/w3c/W3CHttpCommandCodec.java

<li>Removed 12 static imports for storage commands<br> <li> Removed 12 command aliases for storage operations<br> <li> Removed 12 JavaScript implementations for storage methods<br> <li> Eliminated localStorage and sessionStorage script executions


</details>


  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium/pull/16013/files#diff-d3ee2e9bb5279f45d35691b5fe24402e8b9d1b795b293ace7a5e4c4ec144d5f4">+0/-72</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about Qodo Merge usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

iampopovich committed:
Refactor DriverCommand and W3CHttpCommandCodec for improved readability

pujagani commented:
@iampopovich The changes made and the PR title don't seem to align. Can you please update it?

iampopovich committed:
Merge branch 'trunk' into feat-10397/cleanup-html5-leftovers

iampopovich committed:
Merge branch 'trunk' into feat-10397/cleanup-html5-leftovers

iampopovich committed:
Merge branch 'trunk' into feat-10397/cleanup-html5-leftovers

asolntsev reviewed (approved):
I hope there is some replacement for these endpoints? Or how can people clear localStorage etc.?

iampopovich committed:
[java] Deprecate localStorage and sessionStorage commands in DriverCommand and update W3CHttpCommandCodec

iampopovich commented:
@asolntsev @Delta456 thanks for review 
These changes break backward compatibility and will break users' working code if they're clearing localStorage using the old helper methods.

I found a similar pull request in the Ruby client: https://github.com/SeleniumHQ/selenium/pull/14276.  
I reverted my changes and added `@deprecated` annotations with example code. However, it might not be obvious to users that they now need to use the JavaScript executor instead.

asolntsev commented:
> will break users' working code if they're clearing localStorage using the old helper methods

Totally fine for me :) if users clearly understand what should they use instead.

iampopovich committed:
Refactor deprecation comments for localStorage and sessionStorage commands in DriverCommand
